pkg/metrics/metrics.go

```diff
@@ -59,6 +65,67 @@ const (
 	CQStatusTerminating ClusterQueueStatus = "terminating"
 )
 
+type LocalQueueMetricsConfig struct {
+	Enabled            bool
+	NamespaceSelector  *metav1.LabelSelector
+	LocalQueueSelector *metav1.LabelSelector
+}
+
+var lqMetricsConfigSingleton *LocalQueueMetricsConfig
+
+func SetLocalQueueMetrics(lqMetricsConfig *LocalQueueMetricsConfig) {
+	lqMetricsConfigSingleton = lqMetricsConfig
+}
+
+var once sync.Once
+
+func GetLocalQueueMetrics() *LocalQueueMetricsConfig {
+	if lqMetricsConfigSingleton == nil {
+		once.Do(
+			func() {
+				lqMetricsConfigSingleton = &LocalQueueMetricsConfig{}
+			})
+	}
+	return lqMetricsConfigSingleton
+}
+
+func ShouldReportLocalMetrics(c client.Client, q *kueue.LocalQueue) (bool, error) {
```

Comment:
```suggestion
func ShouldReportLocalMetrics(ctx context.Context, c client.Client, q *kueue.LocalQueue) (bool, error) {
```

It would be good to use propagated context.